drivers/net/wireless/ath/ath9k/ptp.c

```diff
@@ -0,0 +1,124 @@
+#include "ath9k.h"
+
+#include <linux/ktime.h>
+#include <linux/clocksource.h>
+#include <linux/ptp_clock_kernel.h>
+
+static int ath9k_phc_adjfine(struct ptp_clock_info *ptp, long scaled_ppm) {
+#if 0
+    struct ath_softc *sc = container_of(ptp, struct ath_softc, ptp_clock_info);
+    unsigned long flags;
+    int neg_adj = 0;
+    u32 mult, diff;
+    u64 adj;
+
+    if (scaled_ppm < 0) {
+        neg_adj = -1;
+        scaled_ppm = -scaled_ppm;
+    }
+    mult = sc->cc_mult;
+    adj = mult;
+    adj *= scaled_ppm;
+    diff = div_u64(adj, 1000000000ULL);
```

Comment:
The mistake here is still using 1,000,000,000 (1 billion). You need to use 1million * 2^16, but you can actually just replace this all with adjust_by_scaled_ppm.